src/fips.c

```diff
@@ -600,6 +605,8 @@ fips_powerup_tests(void)
 		}
 	}
 
+	ica_stats_enabled = stats_mode_temp;
```

Comment:
What if you return early in line 599 ? Should you maybe restore ica_stats_enabled there as well?

Reply:
Argh ... yes, good catch. Done.